consensus/src/marshal/mod.rs

```diff
@@ -447,7 +447,7 @@ mod tests {
                         .iter_mut()
                         .enumerate()
                     {
-                        if (do_finalize && i <= QUORUM as usize)
+                        if (do_finalize && i < QUORUM as usize)
```

Comment:
This was actually a bug in the test; `QUORUM` is `3` & `NUM_VALIDATORS` is `4`, so we'd show all validators the finalization when `do_finalize` was `true` 🫣